scripts/download_pl_metadata.py

```diff
@@ -0,0 +1,79 @@
+from typing import Optional, Any
+
+import typer
+from dotenv import load_dotenv
+from mpire.pool import WorkerPool
+from pymongo import MongoClient
+from pymongo.errors import BulkWriteError
+from pymongo.server_api import ServerApi
+from requests import HTTPError
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, wait_random_exponential
+from loguru import logger
+
+from juddges.data.pl_court_api import PolishCourtAPI
+
+N_JOBS = 8
+BATCH_SIZE = 1_000
+
+load_dotenv("secrets.env", verbose=True)
+
+
+def main(
+    mongo_uri: str = typer.Option(..., envvar="MONGO_URI"),
+    batch_size: int = typer.Option(BATCH_SIZE),
+    n_jobs: int = typer.Option(N_JOBS),
+    limit: Optional[int] = typer.Option(None),
+) -> None:
+    api = PolishCourtAPI()
+    total_judgements = api.get_number_of_judgements()
+    logger.info(f"Total judgements found: {total_judgements}")
+
+    if limit is not None:
+        total_judgements = min(total_judgements, limit)
+
+    offsets = list(range(0, total_judgements, batch_size))
+    download_metadata = MetadataDownloader(mongo_uri, batch_size)
+    with WorkerPool(n_jobs=n_jobs) as pool:
+        pool.map_unordered(download_metadata, offsets, progress_bar=True)
+
+
+class MetadataDownloader:
+    def __init__(self, mongo_uri: str, batch_size: int):
+        self.mongo_uri = mongo_uri
+        self.batch_size = batch_size
+
+    @retry(
+        wait=wait_random_exponential(multiplier=1, max=60),
+        retry=retry_if_exception_type(HTTPError),
+        stop=stop_after_attempt(3),
```

Comment:
as above

Reply:
increased to 5, and changed min/max times